### PR TITLE
927 - Added socket connection limits.

### PR DESCRIPTION
**Pull request by ebadiere**
**Description**:

This PR add socket connection limits to the websocket server.  
* CONNECTION_LIMIT was added to the env.example and localAcceptance.env
* CONNECTION_LIMIT was listed in the README.md

**Notes for reviewer**:
This was tested by setting the CONNECTION_LIMIT to a low value and adding connections.  It was agreed to start with a connection limit of 10.

### CODE REVIEW

File: packages/server/tests/localAcceptance.env [hunk 1]
```diff
@@ -29,8 +29,9 @@ MIRROR_NODE_RETRIES = 3
...
-ETH_CALL_CONSENSUS=true
-CONSENSUS_MAX_EXECUTION_TIME=10000
-TEST_WS_SERVER=false
-SDK_REQUEST_TIMEOUT=10000
+INPUT_SIZE_LIMIT = 1
+ETH_CALL_CONSENSUS = true
+CONSENSUS_MAX_EXECUTION_TIME = 10000
+TEST_WS_SERVER = false
+SDK_REQUEST_TIMEOUT = 10000
+CONNECTION_LIMIT = 10
```
Comment (AlfredoG87) [inline packages/server/tests/localAcceptance.env:37]: Does whitespaces does not affect the values of the properties?

File: packages/ws-server/src/webSocketServer.ts [hunk 2]
```diff
@@ -44,11 +44,22 @@ const relay: Relay = new RelayImpl(logger, register);
...
 
+const MAX_CONNECTIONS = parseInt(process.env.CONNECTION_LIMIT || '10');
+let connectedClients = 0;
+
 app.ws.use((ctx) => {
     ctx.websocket.id = relay.subs().generateId();
     logger.info(`${LOGGER_PREFIX} new connection ${ctx.websocket.id}`);
 
+    if (connectedClients >= MAX_CONNECTIONS) {
+        ctx.websocket.send("Connections exceeded");
```
Comment (AlfredoG87) [inline packages/ws-server/src/webSocketServer.ts:55]: could message be changed to something like: "Max concurrent connections limit exceeded, please try again later"

**Comment by Nana-EC** [comment]
PR should note the ticket it's addressing so that on merge it closes that ticket

File: .env.example [hunk 3]
```diff
@@ -28,4 +28,5 @@ ETH_CALL_CONSENSUS=
 CONSENSUS_MAX_EXECUTION_TIME=
 ETH_CALL_CACHE_TTL=
 WEB_SOCKET_PORT=
-SDK_REQUEST_TIMEOUT=
\ No newline at end of file
+SDK_REQUEST_TIMEOUT=
+CONNECTION_LIMIT=
```
Comment (Nana-EC) [inline .env.example:32]: nit: new line

File: packages/ws-server/src/webSocketServer.ts (hunk 2 above)
Comment (Nana-EC) [inline packages/ws-server/src/webSocketServer.ts:55]: Are we able to acceptance test this ?
Maybe we need to wait for the base and acceptance PR to go in and rebase .

**Review by Nana-EC — COMMENTED** [review]
Looking good. 
Waiting on rebase to main and potential acceptance test

**Comment by ebadiere** [comment]
Applied changes to another commit and PL since the merge of the eth_subscribe work into main.

**Pull request closed without merging**